src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -86,6 +86,32 @@ private static string GetRequestParameterName(RequestParameter requestParameter)
             return language.SerializedName ?? language.Name;
         }
 
+        public CSharpType? GetReturnType(InputOperation operation, string defaultNamespace)
+        {
+            var bodyType = operation.Responses.FirstOrDefault()?.BodyType;
+            if (bodyType != null && bodyType is InputModelType)
+            {
+                return new CSharpType(new ModelTypeProvider((bodyType as InputModelType)!, _typeFactory, defaultNamespace, null));
```

Comment:
Do we current we only support json object as return value, primitive such as string not supported?

Reply:
Correct. We don't support non-user defined type.

Reply:
Why not `_typeFactory.CreateType()`?

Reply:
We can, but it is meaningless. We don't support this scenario and `_typeFactory.CreateType()` will never be hit.

Reply:
This is a sister issue of the https://github.com/Azure/autorest.csharp/issues/2535, but we can address it only after model generation is fully implemented